packages/form-js-viewer/src/render/components/form-fields/Checklist.js

```diff
@@ -0,0 +1,92 @@
+import { useContext } from 'preact/hooks';
+
+import { FormContext } from '../../context';
+
+import Description from '../Description';
+import Errors from '../Errors';
+import Label from '../Label';
+
+import {
+  formFieldClasses,
+  prefixId
+} from '../Util';
+
+const type = 'checklist';
+
+
+export default function Checklist(props) {
+  const {
+    disabled,
+    errors = [],
+    field,
+    value = [],
+  } = props;
+
+  const {
+    description,
+    id,
+    label,
+    values
+  } = field;
+
+  const toggleCheckbox = (v) => {
+
+    var newValue = [...value];
+
+    if (!newValue.includes(v)) {
```

Comment:
I assume this works because we can rely on that UI & data is always in sync, right? So we will technically never have the case.

* Checkbox is checked
* Value is not in the data

What I mean specifically: We do not react on the `event.target.checked` itself, but only toggling the data.

Reply:
Yeah, the rendering state is directly tethered to the underlying data. The reason I did it this way was that I modeled this component from the radio buttons which do it like that.

Whether it's the right thing to do or not, 🤷 I'm not sure. We can also do it how it's done with the checkbox component and react to the checked state. I don't think it matters much.